File: viavem_api/viavem_api/Models/Usuarios.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace viavem_api.Models
{
    public class Usuarios
    {
        [Key]
        public int Id { get; set; }


        [ForeignKey("Passagens")]
        public int IdPassagens { get; set; }

       public Passagens Passagens { get; set; }
       public string Nome { get; set; }
       public string Senha { get; set; }
       public string Email { get; set; }
    }
}

File: front/cliente-api-main/pages/Usuarios.js
import "bootstrap/dist/css/bootstrap.css";
import { useEffect, useState } from "react";
import axios from "axios";
import style from "../styles/Home.module.css";

const User = () => {
  const [users, setUsers] = useState([]);

  useEffect(() => {
    // Faça uma chamada GET para a API Spring Boot
    axios
      .get("https://localhost:7203/api/Usuarios/")
      .then((response) => {
        setUsers(response.data);
      })
      .catch((error) => {
        console.error("Erro ao buscar a lista de Categorias:", error);
      });
  }, []);

  return (
    <div>
      <h1 className={style.h1}>Lista de Usuarios</h1>
      <table className="table container tabela">
        <thead>
          <tr>
            <th>Id</th>
            <th>Id da passagem</th>
            <th>nome</th>
            <th>email</th>
            <th>senha</th>
          </tr>
        </thead>
        {users.map((element) => (
          <tbody key={element.id}>
            <tr className={style.tabela}>
              <td>{element.id}</td>
              <td>{element.idPassagens}</td>
              <td>{element.nome}</td>
              <td>{element.email}</td>
              <td>{element.senha}</td>
            </tr>
          </tbody>
        ))}
      </table>
    </div>
  );
};

export default User;
